iree/modules/hal/hal_module.cc

```diff
@@ -764,13 +764,16 @@ class HALModuleState final {
 
   StatusOr<int32_t> SemaphoreAwait(vm::ref<iree_hal_semaphore_t> semaphore,
                                    uint32_t new_value) {
-    IREE_TRACE_SCOPE0("HALModuleState::SemaphoreAwait");
-    // TODO(benvanik): allow for deadline exceeded returns? We don't allow
-    // setting deadlines now (and when we do in the future it'll be the fiber
-    // manager that handles them), so any failure indicates total failure.
-    IREE_RETURN_IF_ERROR(iree_hal_semaphore_wait_with_deadline(
-        semaphore.get(), new_value, IREE_TIME_INFINITE_FUTURE));
-    return OkStatus();
+    // TODO(benvanik): coroutine magic
+    iree_status_t status = iree_hal_semaphore_wait_with_deadline(
+        semaphore.get(), new_value, IREE_TIME_INFINITE_FUTURE);
+    if (iree_status_is_ok(status)) {
+      return 0;
+    } else if (iree_status_is_deadline_exceeded(status)) {
+      // Propagate deadline exceeded back to the VM.
+      return static_cast<int32_t>(iree_status_consume_code(status));
+    }
+    return std::move(status);
```

Comment:
This diff was lifted from https://github.com/google/iree/pull/3091 and builds on my Windows machine with CMake but not [on the CI](https://source.cloud.google.com/results/invocations/9a1c1006-e788-48ec-a9e5-3d6fe6b97700/targets/iree%2Fgcp_ubuntu%2Fbazel%2Flinux%2Fx86%2Fcore%2Fpresubmit/log) (?). I can just wait for that PR to be merged first.

Reply:
PR has been updated and now builds on CI

Reply:
Rebased on top of that commit and fixed the function info logging.